### PR TITLE
`prefer-dom-node-dataset`: Check `.hasAttribute()` and `.getAttribute()`

### LINKED ISSUE #1669: Check `.hasAttribute()` in `prefer-dom-node-dataset` rule
[MDN suggest using `in` operator to check existence](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#accessing_values), I'm not sure what's the best, maybe `hasOwnProperty()`?

## Fail

```js
element.hasAttribute('data-foo');
```

## Pass

```js
"foo" in element.dataset;
```

### PR DESCRIPTION
**Pull request by fisker**
Fixes #1669

I just realized that we are not checking `.getAttribute()`, any reason behind this? I need a break...

### CODE REVIEW

File: rules/prefer-dom-node-dataset.js [hunk 1]
```diff
@@ -54,7 +68,7 @@ module.exports = {
 	meta: {
 		type: 'suggestion',
 		docs: {
-			description: 'Prefer using `.dataset` on DOM elements over `.setAttribute(…)` and `.removeAttribute(…)`.',
+			description: 'Prefer using `.dataset` on DOM elements over accessing attributes.',
```
Comment (sindresorhus) [inline rules/prefer-dom-node-dataset.js:71]:
```suggestion
			description: 'Prefer using `.dataset` on DOM elements over calling attribute methods.',
```

?

**Comment by sindresorhus** [comment]
> I just realized that we are not checking .getAttribute(), any reason behind this? I need a break...

No reason other than it not being thought of when the rule was initially added: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/166

File: rules/prefer-dom-node-dataset.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ const create = context => ({
 			}
 
 			case 'hasAttribute':
+				// TODO: Use `Object.hasOwn()` here when targeting Node.js 16.
```
Comment (fisker) [inline rules/prefer-dom-node-dataset.js:49]: Arr you sure you want this comment? This code should running in browser.

Comment (sindresorhus) [reply rules/prefer-dom-node-dataset.js:49]: `Object.hasOwn()` is already in Chrome, Safari, and Firefox.

**Pull request merged**